x/bank/types/balance.go

```diff
@@ -64,12 +65,18 @@ func SanitizeGenesisBalances(balances []Balance) []Balance {
 
 	// 1. Retrieve the address equivalents for each Balance's address.
 	addresses := make([]sdk.AccAddress, len(balances))
+	// 2. Track any duplicate addresses to avoid false positives on invariant checks.
+	seen := make(map[string]struct{})
 	for i := range balances {
 		addr, _ := sdk.AccAddressFromBech32(balances[i].Address)
 		addresses[i] = addr
+		if _, exists := seen[string(addr)]; exists {
+			panic(fmt.Errorf("genesis state has a duplicate account: %q", balances[i].Address))
```

Comment:
Going into panic, it doesn't have to from fmt.Errorf, it could be just from fmt.Sprintf but also to show more context, earlier I had suggested and perhaps let's do this:

```go
if _, exists := seen[string(add)]; exists {
   panic(fmt.Sprintf("Genesis state has duplicate account: %q aka %x", balances[i].Address, addr))
```

Reply:
Oh, you suggested `%q aka %x", addr, addr` so I thought it was a typo. Fixed.